test/integration/scenarios/network/peer_blacklist.js

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const fs = require('fs');
+const childProcess = require('child_process');
+const Peer = require('../../../../logic/peer');
+const utils = require('../../utils');
+
+const totalPeers = 10;
+// Each peer connected to 9 other pairs and have 2 connection for bi-directional communication
+const expectedOutgoingConnections = (totalPeers - 1) * totalPeers * 2;
+const expectedConnectionsAfterBlacklisting = (totalPeers - 2) * (totalPeers - 1) * 2;
+
+module.exports = params => {
+	describe('Peer Blacklisted', () => {
+		const getAllPeers = sockets => {
+			return Promise.all(
+				sockets.map(socket => {
+					if (socket.state === 'open') {
+						return socket.call('list', {});
+					}
+				})
+			);
+		};
+
+		const restartNode = nodeName => {
+			return childProcess.execSync(`pm2 restart ${nodeName}`);
+		};
+
+		const wsPorts = new Set();
+
+		describe('when peers are mutually connected in the network', () => {
+			before(() => {
+				return getAllPeers(params.sockets).then(mutualPeers => {
+					mutualPeers.forEach(mutualPeer => {
+						if (mutualPeer) {
+							mutualPeer.peers.map(peer => {
+								wsPorts.add(peer.wsPort);
+								expect(peer.state).to.be.eql(Peer.STATE.CONNECTED);
+							});
+						}
+					});
+				});
+			});
+
+			it(`there should be ${expectedOutgoingConnections} established connections from 500[0-9] ports`, done => {
+				utils.getEstablishedConnections(
+					Array.from(wsPorts),
+					(err, numOfConnections) => {
+						if (err) {
+							return done(err);
+						}
+
+						if ((numOfConnections - 20) <= expectedOutgoingConnections) {
```

Comment:
Please add a comment line here stating why you subtracted 20 from number of connections.